packages/editor/src/components/post-format/index.js

```diff
@@ -42,20 +44,17 @@ export default function PostFormat() {
 	const instanceId = useInstanceId( PostFormat );
 	const postFormatSelectorId = `post-format-selector-${ instanceId }`;
 
-	const { postFormat, suggestedFormat, supportedFormats } = useSelect(
-		( select ) => {
-			const { getEditedPostAttribute, getSuggestedPostFormat } =
-				select( editorStore );
-			const _postFormat = getEditedPostAttribute( 'format' );
-			const themeSupports = select( coreStore ).getThemeSupports();
-			return {
-				postFormat: _postFormat ?? 'standard',
-				suggestedFormat: getSuggestedPostFormat(),
-				supportedFormats: themeSupports.formats,
-			};
-		},
-		[]
-	);
+	const { postFormat, blocks, supportedFormats } = useSelect( ( select ) => {
+		const { getEditedPostAttribute } = select( editorStore );
+		const _postFormat = getEditedPostAttribute( 'format' );
+		const themeSupports = select( coreStore ).getThemeSupports();
+		return {
+			postFormat: _postFormat ?? 'standard',
+			blocks: select( blockEditorStore ).getBlocks(),
+			supportedFormats: themeSupports.formats,
+		};
+	}, [] );
+	const suggestedFormat = getSuggestedPostFormat( blocks );
```

Comment:
It might be more performant to derive value inside the `mapSelect`, so we don't re-render the component on every block change.

Reply:
Right, will have to revert to using the selector